### PR TITLE
FEAT: Added integrations for devtron

### PR DESCRIPTION
**Pull request by dheeth**
# Description

This PR has changes for argo-cd integration in devtron-operator chart as well as inception

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Tested with fresh installations multiple times
- [x] Tested with upgrades on existing installations

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: charts/devtron/README.md [hunk 1]
```diff
@@ -51,10 +51,20 @@ $ kubectl apply -n devtroncd -f charts/template/devtron-installer.yaml
 
 devtron dashboard in now available at the `BASE_URL/dashboard`, where `BASE_URL` is same as provided in `values.yaml` in case of installation via helm chart OR provided in `charts/template/configmap-secret.yaml` in case of installation via kubectl.
 
-For login use username:`admin` and for password run command mentioned below.
+#### For Devtron version v0.6.0 and higher
```
Comment (nishant-d) [inline charts/devtron/README.md:54]: how to check my version ?

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: charts/devtron/templates/dashboard.yaml [hunk 2]
```diff
@@ -7,6 +8,10 @@ metadata:
     release: devtron
   annotations:
     "helm.sh/resource-policy": keep
+    {{- if and ($argocdEnabled.enabled) ($.Release.IsInstall) }}
+    "helm.sh/hook": pre-install
```
Comment (prakarsh-dt) [inline charts/devtron/templates/dashboard.yaml:12]: To be removed as not needed

File: charts/devtron/values.yaml [hunk 3]
```diff
@@ -105,3 +104,151 @@ components:
...
+                app.kubernetes.io/name: argocd-redis
+            topologyKey: kubernetes.io/hostname
+          weight: 100
+        - podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/part-of: argocd
+            topologyKey: kubernetes.io/hostname
+          weight: 5
+
```
Comment (prakarsh-dt) [inline charts/devtron/values.yaml:181]: Included comments for better readability

File: manifests/installation-script [hunk 4]
```diff
@@ -637,15 +507,14 @@ if !helmInstallation {
...
-log("executed guard setup");
+
 if !helmInstallation {
   dashboard = kubectl apply -n devtroncd dashboard -u dashboardOverride;
   log("executed dashboard setup");
 }
 gitSensor = kubectl apply -n devtroncd gitSensor -u gitSensorOverride;
 log("executed git sensor setup");
-imageScanner = kubectl apply -n devtroncd imageScanner -u imageScannerOverride;
+##imageScanner = kubectl apply -n devtroncd imageScanner -u imageScannerOverride;
```
Comment (prakarsh-dt) [inline manifests/installation-script:517]: Remove

File: manifests/yamls/app-sync-labels.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    kubectl annotate rolebinding argo-binding -n argo "meta.helm.sh/release-name=$RELEASE_NAME" "meta.helm.sh/release-namespace=devtroncd"
+    kubectl label cm workflow-controller-configmap -n argo "app.kubernetes.io/managed-by=Helm"
+    kubectl annotate cm workflow-controller-configmap -n argo "meta.helm.sh/release-name=$RELEASE_NAME" "meta.helm.sh/release-namespace=devtroncd"
+    kubectl label deploy workflow-controller -n argo "app.kubernetes.io/managed-by=Helm"
+    kubectl annotate deploy workflow-controller -n argo "meta.helm.sh/release-name=$RELEASE_NAME" "meta.helm.sh/release-namespace=devtroncd"
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: app-sync-labels
```
Comment (prakarsh-dt) [inline manifests/yamls/app-sync-labels.yaml:88]: change the name to something generic devtron-housekeeping

File: charts/devtron/README.md (hunk 1 above)
Comment (dheeth) [reply charts/devtron/README.md:62]: Added steps to check current version

File: charts/devtron/templates/dashboard.yaml (hunk 2 above)
Comment (dheeth) [reply charts/devtron/templates/dashboard.yaml]: Removed

File: manifests/installation-script (hunk 4 above)
Comment (dheeth) [reply manifests/installation-script]: Removed

File: manifests/yamls/app-sync-labels.yaml (hunk 5 above)
Comment (dheeth) [reply manifests/yamls/app-sync-labels.yaml]: Renamed to devtron-housekeeping

File: charts/devtron/values.yaml (hunk 3 above)
Comment (dheeth) [reply charts/devtron/values.yaml]: Comments added

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**